frontend/components/ContributionsRow.tsx

```diff
@@ -117,14 +120,17 @@ export const ContributionsRow = ({
         }
 
         setArchives((prevArchives) => {
-          let all = [
-            ...(Array.isArray(prevArchives) ? prevArchives : []),
-            ...newArchives,
-          ];
-          // unique by id
-          all = all.filter(
-            (v, i, a) => a.findIndex((t) => t.id === v.id) === i,
-          );
+          const all = [...(Array.isArray(prevArchives) ? prevArchives : [])];
+          for (const archive of newArchives) {
+            const oldIndex = all.findIndex((t) => t.id === archive.id);
+            if (oldIndex !== -1) {
+              all[oldIndex] = {
+                ...archive,
+              };
```

Comment:
I'm confused about this. can you elaborate why?

Reply:
new archive object might have same id but updated properties for who the owner is and whether it's minted of not @Ledespana